presto-main/src/main/java/com/facebook/presto/execution/CreateViewTask.java

```diff
@@ -93,8 +96,12 @@ public String explain(CreateView statement)
 
         List<ViewColumn> columns = analysis.getOutputDescriptor()
                 .getVisibleFields().stream()
-                .map(field -> new ViewColumn(field.getName().get(), field.getType()))
-                .collect(toImmutableList());
+                .map(field -> {
```

Comment:
Rather than overloading `map`, using `peek` for validations is a bit cleaner:
```java
.peek(field -> {
    if (field.getType().equals(UnknownType.UNKNOWN)) {
        throw new SemanticException(NOT_SUPPORTED, statement, "View column type is unknown: %s", field.getName());
    }
})